AppLovinAdapter/src/main/java/com/chartboost/mediation/applovinadapter/AppLovinAdapter.kt

```diff
@@ -52,79 +52,115 @@ class AppLovinAdapter : PartnerAdapter {
 
                     adInfo?.let { adId ->
                         withContext(Main) {
-                            AppLovinSdk.getInstance(context).settings.testDeviceAdvertisingIds =
-                                listOf(adId)
+                            appLovinSdk?.let { sdk ->
+                                sdk.settings.testDeviceAdvertisingIds = listOf(adId)
+
+                                PartnerLogController.log(
+                                    CUSTOM,
+                                    "AppLovin test mode is enabled. Remember to disable it before publishing."
+                                )
+                            } ?: run {
+                                PartnerLogController.log(
+                                    CUSTOM,
+                                    "Unable to set test mode. AppLovin SDK instance is null."
+                                )
+                            }
                         }
                     } ?: run {
-                        PartnerLogController.log(
-                            CUSTOM,
-                            "AppLovin test mode is disabled. No advertising id found."
-                        )
-                        AppLovinSdk.getInstance(context).settings.testDeviceAdvertisingIds =
-                            emptyList()
+                        appLovinSdk?.let { sdk ->
+                            sdk.settings.testDeviceAdvertisingIds = emptyList()
+
+                            PartnerLogController.log(
+                                CUSTOM,
+                                "AppLovin test mode is disabled. No advertising id found."
+                            )
+                        } ?: run {
+                            PartnerLogController.log(
+                                CUSTOM,
+                                "Unable to set test mode. AppLovin SDK instance is null."
+                            )
+                        }
                     }
                 }
             } else {
-                AppLovinSdk.getInstance(context).settings.testDeviceAdvertisingIds = emptyList()
-            }
+                appLovinSdk?.let { sdk ->
```

Comment:
you also aren't the one who did this, but man it would save a lot of lines by doing
`val appLovinSdk = appLovinSdk ?: run { log some stuff then return }`

Reply:
yeah let me try something